src/Microsoft.Extensions.Configuration.AzureAppConfiguration/RequestTracingOptions.cs

```diff
@@ -68,13 +70,57 @@ internal class RequestTracingOptions
         /// </summary>
         public bool IsPushRefreshUsed { get; set; } = false;
 
+        /// <summary>
+        /// Flag to indicate whether any key-value uses the json content type and contains
+        /// a parameter indicating an AI profile.
+        /// </summary>
+        public bool UsesAIConfiguration { get; set; } = false;
+
+        /// <summary>
+        /// Flag to indicate whether any key-value uses the json content type and contains
+        /// a parameter indicating an AI chat completion profile.
+        /// </summary>
+        public bool UsesAIChatCompletionConfiguration { get; set; } = false;
+
+        /// <summary>
+        /// Resets the AI configuration tracing flags.
+        /// </summary>
+        public void ResetAiConfigurationTracing()
+        {
+            UsesAIConfiguration = false;
+            UsesAIChatCompletionConfiguration = false;
+        }
+
+        /// <summary>
+        /// Updates AI configuration tracing flags based on the provided content type.
+        /// </summary>
+        /// <param name="contentTypeString">The content type to analyze.</param>
+        public void UpdateAiConfigurationTracing(string contentTypeString)
+        {
+            if (!string.IsNullOrWhiteSpace(contentTypeString) && contentTypeString.TryParseContentType(out ContentType contentType))
```

Comment:
If `UsesAIChatCompletionConfiguration` is already true there's no need to do anything.